app/controllers/discourse_custom_topic_lists/lists_controller.rb

```diff
@@ -3,9 +3,41 @@
 module ::DiscourseCustomTopicLists
   class ListsController < ::ApplicationController
     requires_plugin PLUGIN_NAME
+    skip_before_action :check_xhr, only: [:lists_feed]
 
     def show
       render json: success_json
     end
+
+    def lists_feed
+      discourse_expires_in 1.minute
+
+      raise Discourse::NotFound if !SiteSetting.experimental_topics_filter
+      raise Discourse::InvalidParameters.new(:topic_list_name) if !params[:topic_list_name]
+      topic_query_opts = { no_definitions: !SiteSetting.show_category_definitions_in_topic_lists }
+
+      list_item =
+        DiscourseCustomTopicLists::CustomTopicList.new(current_user).find_by_slug(
```

Comment:
Sorry I don't mean to drag this on but I still don't think we are quite there yet in terms of test coverage especially when the custom topic lists uses groups as access control. We are relying on `DiscourseCustomTopicLists::CustomTopicList#lists` to ensure that the user belongs in the right groups before they can view a custom topic list. However, I do not see any test coverage for `DiscourseCustomTopicLists::CustomTopicList#lists` and the request specs added in this PR does not consider access control as well. If we do not have proper test coverage here, we are increasing our risk of a security issue in the future.

Reply:
Sure! Thanks a lot for being so thorough with this PR!

I've added a few tests. If you have more suggestions, I'm super happy to hear them!